hdk/virtual_machine/syntax.py

```diff
@@ -1,8 +1,61 @@
 """Defines dataclasses that represent different types of vm commands."""
+import re
 from dataclasses import dataclass
 from typing import ClassVar, TypeAlias
 
 
+@dataclass(frozen=True)
+class BranchingCommand:
+    """Represents a Branching command in a virtual machine.
+
+    Attributes:
+        operation: The type of branching operation to be performed.
+        label_name: The symbol defined by the Label.
+    """
+
+    _ALLOWED_OPERATIONS: ClassVar[set[str]] = {"label", "goto", "if-goto"}
+
+    operation: str
+    label_name: str
+
+    def __post_init__(self):
+        cls = self.__class__
+        if not _is_symbol_valid(self.label_name):
+            raise ValueError(f"Invalid symbol {self.label_name!r} for Label.")
+        if self.operation not in cls._ALLOWED_OPERATIONS:
+            raise ValueError(
+                f"Wrong operation {self.operation!r} for branching command."
+            )
+
+
+@dataclass(frozen=True)
+class FunctionCommand:
```

Comment:
Split the FunctionCommand into 3 different classes:
1. ReturnCommand
2. FunctionDefinitionCommand
3. FunctionCallCommand
because all three types of commands have different structure and should be treated separately.